File: regexlib/python_re_test_file/regexlib_327.py
# 327
# T(?<action>\w)?-(?<user>\w+.+\d)=(?<time>\d+(?:\.\d+)?)
# POLYNOMIAL
# nums:4
# POLYNOMIAL AttackString:"T-"+"1"*10000+"! _1SLQ_2"

import re
from time import perf_counter

regex = """T(?<action>\w)?-(?<user>\w+.+\d)=(?<time>\d+(?:\.\d+)?)"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "T-" + "1" * i * 10000 + "! _1SLQ_2"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")

File: regexlib/javascript_test_file/regexlib_327.js
/* 327
 * T(?<action>\w)?-(?<user>\w+.+\d)=(?<time>\d+(?:\.\d+)?)
 * POLYNOMIAL
 * nums:3
 * POLYNOMIAL AttackString:"T-"+"1"*10000+"! _1SLQ_2"
 */
var REGEX = /T(?<action>\w)?-(?<user>\w+.+\d)=(?<time>\d+(?:\.\d+)?)/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = 'T-' + '1'.repeat(i*10000) + '! _1SLQ_2'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 10000 + ": " + time_cost+" ms")
}